File: app/src/main/java/com/halfplatepoha/jisho/history/HistoryAdapter.java
package com.halfplatepoha.jisho.history;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.halfplatepoha.jisho.R;
import com.halfplatepoha.jisho.analytics.Analytics;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by surjo on 22/04/17.
 */

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder>{

    private ArrayList<String> history;
    private OnSearchClickedListener searchClickedListener;
    private LayoutInflater inflater;

    public HistoryAdapter(Context context) {
        inflater = LayoutInflater.from(context);
    }

    public void setSearchClickedListener(OnSearchClickedListener searchClickedListener) {
        this.searchClickedListener = searchClickedListener;
    }

    @Override
    public HistoryViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View historyView = inflater.inflate(R.layout.row_history, parent, false);
        return new HistoryViewHolder(historyView);
    }

    @Override
    public void onBindViewHolder(HistoryViewHolder holder, int position) {
        holder.tvHistory.setText(history.get(position));
    }

    @Override
    public int getItemCount() {
        if(history != null)
            return history.size();
        return 0;
    }

    public void addHistory(String historyString) {
        if(history == null)
            history = new ArrayList<>();
        int position = history.size();
        history.add(historyString);

        notifyItemInserted(position);
    }

    public class HistoryViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.tvHistory)
        TextView tvHistory;

        public HistoryViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        @OnClick(R.id.history)
        public void historyClicked() {
            Analytics.getInstance().recordClick("History_Search", history.get(getAdapterPosition()));
            if(searchClickedListener != null)
                searchClickedListener.onSearchClicked(history.get(getAdapterPosition()));
        }
    }

    public interface OnSearchClickedListener {
        void onSearchClicked(String searchString);
    }
}

File: app/src/main/java/com/halfplatepoha/jisho/v2/history/HistoryAdapter.kt
package com.halfplatepoha.jisho.v2.history

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.halfplatepoha.jisho.R
import com.halfplatepoha.jisho.db.v2.HistoryV2

class HistoryAdapter: RecyclerView.Adapter<HistoryViewHolder>() {

    var historyItems: List<HistoryV2>? = null

    var listener: ((id: String) -> Unit)? = null
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HistoryViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_history, parent, false)
        return HistoryViewHolder(view, listener)
    }

    override fun getItemCount(): Int {
        return historyItems?.size ?: 0
    }

    override fun onBindViewHolder(holder: HistoryViewHolder, position: Int) {
        historyItems?.get(position)?.run { holder.bind(this) }
    }

}

class HistoryViewHolder(view: View, val listener: ((id: String) -> Unit)?): RecyclerView.ViewHolder(view) {

    fun bind(history: HistoryV2) {
        itemView.findViewById<TextView>(R.id.tvHistory).text = history.word
        itemView.tag = history.id
        itemView.setOnClickListener { listener?.apply { it.tag?.toString() } }
    }

}